app/bundles/FormBundle/Model/SubmissionModel.php

```
@@ -406,7 +409,11 @@ public function saveSubmission($post, $server, Form $form, Request $request, $re
 
         // Create/update lead
         if (!empty($leadFieldMatches)) {
-            $lead = $this->createLeadFromSubmit($form, $leadFieldMatches, $leadFields);
+            try {
+                $lead = $this->createLeadFromSubmit($form, $leadFieldMatches, $leadFields);
+            } catch (\Exception $exception) {
+                return ['errors' => $exception->getMessage()];
+            }
```

Similar catch for all exceptions is on several places in Mautic. It's hiding errors from the logs or removes stack trace so no one can investigate where the error comes from.

Suggestion: Catch only the specific exception that you want to display to the user and let other exceptions fail hard and logged.
2.